pkg/apis/kubeone/types.go

```diff
@@ -111,6 +111,30 @@ type ClusterNetworkConfig struct {
 	ServiceSubnet     string `json:"serviceSubnet"`
 	ServiceDomainName string `json:"serviceDomainName"`
 	NodePortRange     string `json:"nodePortRange"`
+	CNI               *CNI   `json:"cni,omitempty"`
+}
+
+// CNIProvider type
+type CNIProvider string
+
+// List of CNI Providers
+const (
+	// https://docs.projectcalico.org/v3.7/getting-started/kubernetes/installation/flannel
```

Comment:
Please use a valid godoc comment. Something like `CNIProviderCanal is a Canal CNI plugin (Flannel + Calico). This CNI doesn't support encryption. <link>`.